### DISCUSSION TITLE
converter

### DISCUSSION THREAD

**Discussion by freemant2000** [elimintz/justpy#330] [Ideas]
Hi,

Thanks you for your great work!

Is there any plan to add a converter to the model handling? Currently the model works only if the value in the domain object is a string.

**Comment by elimintz** [comment]
Hi,

I am not sure what you are asking. Could you please elaborate?

**Reply by freemant2000** [reply]
I mean something like below:
```python
class Product:
  def __init__(self, id, name, price):
    self.id=id
    self.name=name
    self.price=price

def p1():
  wp=WebPage()
  wp.pr=Product(1, "pen", 3)
  wp.data=vars(wp.pr)
  wp.inp1=Input(model=[wp, "name"], a=wp)
  wp.inp2=Input(value=[wp, "price"], a=wp)
  return wp
```
The price should be a number, but the input field will put a string into the price variable. To solve this problem, a model might accept a converter that should at least convert a string into the desired data type like:
```
  converter=int  #this is enough for the case on hand
  wp.inp2=Input(value=[wp, "price", converter], a=wp)
```
For simple cases, str() could be used to convert the model value to a string. Perhaps for more complex cases another function is required.

**Reply by elimintz** [reply]
Thanks, I see what you mean. In this simple case, if you define the type of the input as number, it will do the conversion for you:
```python
wp.inp2=Input(value=[wp, "price"], a=wp, type="number")
```

**Reply by freemant2000** [reply]
Thanks! didn't know that :-) 

That would also bring up another issue: what if the conversion fails (e.g., when the input is empty)? Currently it just raises an exception.
